src/number_theory/legendre_symbol/auxiliary.lean

```diff
@@ -30,13 +33,184 @@ begin
     { exact nat.odd_of_mod_four_eq_three h }, },
 end
 
+-- Auxiliary stuff for monoids
+
+/-- If `x ^ n = 1`, then `x ^ m` is the same as `x ^ (m % n)` -/
+lemma pow_eq_pow_mod {M : Type*} [monoid M] {x : M} (m n : ℕ) (h : x ^ n = 1) :
```

Comment:
This should really go in a more basic file, something about monoids, perhaps `group_theory/order_of_element.lean` even though strictly it doesn't need anything from that file. It should also be `to_additive`d

Reply:
`n` could also be left implicit

Reply:
The plan is to collect all these useful lemmas I need in this file, and once I'm done with Gauss sums etc., I'll move them to wherever they fit best. (So that I don't have to touch a lot of files that are close to the root of the import tree while I work on this.)
Regarding to_additive, I assume I just add `@[to_additive]` before the lemma?

Reply:
Yes; this produces `nsmul_eq_nsmul_mod`.

Reply:
By while you work on this do you mean over the next few days/weeks, or only in this PR?
Because lemmas like this one seem useful, and once they are reviewed it seems quite unlikely they will need changing, so if the PR is ready for review I would say it should go in the right place already.

Reply:
`n` could also be left implicit: done.

Reply:
Depends on how quickly the following PRs will be accepted. Hopefully, a few days.

Reply:
As for annoyance with working early in the import heierachy, I agree it can be annoying waiting for things either on your own computer or on CI and sympathize of course, have you seen https://leanprover-community.github.io/tips_and_tricks.html#old-mode this can sometimes reduce the annoyance.

Reply:
I would think it is more efficient to have to endure the annoyance only once (when moving everything in one go) rather than every time I want to add something.

Reply:
Ok but you should add some notes saying this for each lemma, basically every reviewer will look at these lemmas and immediately think "these are in the wrong place", so you can save yourself repeating this conversation by adding "-- TODO move to blah/blah.lean" to every lemma you plan to move later

Reply:
I'll do that when/if more such comments are made.